doc/adr/0018-synchronising-queries-across-indexers-running-in-parallel.rst

```diff
@@ -0,0 +1,65 @@
+
+
+Argument
+--------
+
+A fairly simple solution to the synchronisation problem would be to qualify responses by the current slot number. This is a reasonable first thought, but it does have a high probability of returning results that are qualified by different slot numbers (if the indexers are fully synchronised, it would be a small offset, but still dependent on the various indexers processing speeds). The advantage of this solution is that at least, we now know that the indexers were not fully synchronised. If the application does not require them to be, then this is a reasonable solution.
+
+If, however, the application requires all queries to run at the same slot number, this solution will simply not function. What we could do is request multiple query responses from indexers in the hope that we can find a common slot number where all indexers have an answer. Is this hope realistic? I would argue that it is realistic for indexers that have been fully synchronised with the blockchain since the processed slot offset between them should be fairly small.
+
+Now that will solve our functional problem, but it may introduce some non-functional ones. First, the query results may be fairly big, and returning more of them will put pressure on the network and the parsing of data by the client applications. This can be partially mitigated by selecting a smaller interval, but maybe we can help out here by improving the design a bit. One observation is that we don't actually need to send the full query results. We need to send a query result for the oldest slot number available, but for the newer ones, we only need to send the difference recorded by the next slot. This implementation maps very well to the way we store information in the indexer (an accumulator and a set of events).
+
+Implications
+------------
+
+However, this solution also complicates the life of our users. Now, they need to also have some function that allows them to fold these differences into the older query responses. I would suggest a couple of strategies that we could employ to simplify their lives, that are not the object of this ADR:
+
+1. Write a set of property-based tests that validate that the folding function is correct and instruct our users to verify their implementation using our acceptance tests.
+2. Develop a new software service that can synchronise queries across several indexers. This service would use our implementation of the folding function (which is verified by our tests) and after receiving all the query responses will select the best ones and provide them to the users already folded.
+
+The only thing remaining is to note that if we want to protect against rollbacks we will also need to store the block hash together with the slot number and at this point our design is complete.
```

Comment:
I would add an _Alternative_ section explaining why we have just make sure that all indexers index at the same speed (like we currently have in `marconi-chain-index`) and explain downsides of this approach vs what you're proposing.

Reply:
I moved the `Currently...` paragraph to an alternatives section.